common/namespace/registry.go

```diff
@@ -227,6 +228,17 @@ func (r *registry) GetCacheSize() (sizeOfCacheByName int64, sizeOfCacheByID int6
 	return int64(r.cacheByID.Size()), int64(r.cacheNameToID.Size())
 }
 
+func (r *registry) InvalidateCacheKey(name Name, id ID) {
+	r.cacheLock.Lock()
+	r.cacheByID.Delete(id)
```

Comment:
what about cacheNameToID?

Reply:
The Name to ID mapping should never changed, so we do not need to invalidate it.